wporg-gp-translation-events.php

```diff
@@ -96,8 +99,59 @@ public function activate() {
 		PRIMARY KEY (`translate_event_actions_id`),
 		UNIQUE KEY `event_per_translated_original_per_user` (`event_id`,`locale`,`original_id`,`user_id`)
 		) COMMENT='Tracks translation actions that happened during a translation event'";
-		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
-		dbDelta( $create_table );
+		dbDelta( $create_actions_table );
+
+		// Create attendees table.
+		$create_attendees_table = "
+		CREATE TABLE `{$gp_table_prefix}event_attendees` (
+			`translate_event_attendees_id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+			`event_id` int(10) NOT NULL COMMENT 'Post_ID of the translation_event post in the wp_posts table',
+			`user_id` int(10) NOT NULL COMMENT 'ID of the user who is attending the event',
+		PRIMARY KEY (`translate_event_attendees_id`),
+		UNIQUE KEY `event_per_user` (`event_id`,`user_id`),
+		INDEX `user` (`user_id`)
+		) COMMENT='Attendees of events'";
+		dbDelta( $create_attendees_table );
+
+		// TODO: Remove this once it has been run in production.
+		try {
+			// Don't run it during tests.
+			if ( 'true' === getenv( 'TRANSLATION_EVENTS_NO_IMPORT_LEGACY_ATTENDEES' ) ) {
+				$this->import_legacy_attendees();
+			}
+		} catch ( Exception $e ) {
+			// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
+			error_log( $e );
+		}
+	}
+
+	/**
+	 * Previously, event attendance was tracked through user_meta.
+	 * This function imports this legacy attendance information into the attendees table.
+	 *
+	 * Instead of looping through all users, we consider only users who have contributed to an event.
+	 *
+	 * @deprecated TODO: Remove this function once this has been run in production.
+	 * @throws Exception
+	 */
+	private function import_legacy_attendees(): void {
+		require_once __DIR__ . '/includes/attendee-repository.php';
+		require_once __DIR__ . '/includes/stats-calculator.php';
+
+		$query = new WP_Query(
+			array(
+				'post_type' => self::CPT,
+			)
+		);
+
+		$events              = $query->get_posts();
+		$stats_calculator    = new Stats_Calculator();
+		$attendee_repository = new Attendee_Repository();
+		foreach ( $events as $event ) {
+			foreach ( $stats_calculator->get_contributors( $event ) as $user ) {
```

Comment:
The `get_contributors` method receive and `event id`, so you need to update the parameter passed.

```suggestion
			foreach ( $stats_calculator->get_contributors( $event->ID ) as $users ) {
```
This method returns a list of users, so I think you need to loop though the users and add all to the new table.

Reply:
Great catch, thank you! Fixed.

Reply:
> This method returns a list of users, so I think you need to loop though the users and add all to the new table.

The `foreach` iterates over the result of the call to `get_contributors()`, and assigns the value at that index to the `$user` variable. I'm guessing you missed the `foreach`? Or I might be misunderstanding something.